lib/cli.nix

```diff
@@ -1,6 +1,47 @@
 { lib }:
 
 rec {
+  /* Format a boolean value to a command line list. If boolean is false,
+     returns empty list. Otherwise, returns a list with option name.
+
+     Example:
+       cli.mkDefaultBool "help" true => ["--help"]
+       cli.mkDefaultBool "verbose" false => []
+  */
+  mkDefaultBool = k: v: lib.optional v (mkDefaultOptionName k);
+
+  /* Format a list value to a command line list. Interleaves the same option
+     name for each value in the list.
+
+     Example:
+       cli.mkDefaultList "name" ["ethel" "joe"] => ["--name" "ethel" "--name" "joe"]
+  */
+  mkDefaultList =  k: v: lib.concatMap (mkDefaultOption k) v;
+
+  /* Format a value to a command line list. Returns option name, followed by
+     stringified representation of the value.
+
+     Example:
+       cli.mkDefaultOption "disk" "/dev/sda" => ["--disk" "/dev/sda"]
+       cli.mkDefaultOption "ram" 2048 => ["--ram" "2048"]
+  */
+  mkDefaultOption =
+    k: v: if v == null
+      then []
+      else [ (mkDefaultOptionName k) (lib.generators.mkValueStringDefault {} v) ];
```

Comment:
you are referencing `mkDefaultOptionName` here directly from the let (definition below).

Instead, pass it as argument to the function.

```
mkDefaultOption = { mkOptionName ? mkDefaultOptionName }: k: v: 
```